pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1236,7 +1238,30 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                     });
 
                     schemaVersionFuture.thenAccept(schemaVersion -> {
-                        topic.checkIfTransactionBufferRecoverCompletely(isTxnEnabled).thenAccept(future -> {
+                        topic.checkIfTransactionBufferRecoverCompletely(isTxnEnabled).thenCompose(future -> {
+                            if (Strings.isNullOrEmpty(initialSubscriptionName)
+                                    || topic.getSubscriptions().containsKey(initialSubscriptionName)
+                                    || !topic.isPersistent()) {
+                                return CompletableFuture.completedFuture(null);
+                            }
+                            return isTopicOperationAllowed(
+                                    topicName, TopicOperation.SUBSCRIBE
+                            ).thenCompose(canSubscribe -> {
+                                if (!canSubscribe) {
+                                    log.warn(
+                                            "[{}] Could not create the initial subscription {} for topic {}. Client "
+                                                    + "is not authorized to subscribe with role {}",
+                                            remoteAddress, initialSubscriptionName, topicName, getPrincipal());
+                                    return CompletableFuture.completedFuture(null);
```

Comment:
Should complete with exception?

Reply:
I added the `ProducerInitSubAuthorizationException` to handle this.